src/components/Results.js

```diff
@@ -47,36 +57,85 @@ const styles = (theme) => ({
 });
 
 class Results extends Component {
+  state = {
+    collapsePipeline: [],
+    newResults: [],
+  };
+
   static propTypes = {
     history: PropTypes.shape({
       push: PropTypes.func.isRequired,
     }).isRequired,
     dispatchGetResults: PropTypes.func.isRequired,
     dispatchDeleteResult: PropTypes.func.isRequired,
     dispatchGetAlgorithms: PropTypes.func.isRequired,
+    dispatchGetPipelines: PropTypes.func.isRequired,
     classes: PropTypes.shape({
       addButton: PropTypes.string.isRequired,
       infoAlert: PropTypes.string.isRequired,
     }).isRequired,
     t: PropTypes.func.isRequired,
     results: PropTypes.instanceOf(List),
     algorithms: PropTypes.instanceOf(List),
+    pipelines: PropTypes.instanceOf(List),
     activity: PropTypes.bool.isRequired,
     folder: PropTypes.string,
   };
 
   static defaultProps = {
     results: List(),
     algorithms: List(),
+    pipelines: List(),
     folder: null,
   };
 
   componentDidMount() {
-    const { dispatchGetResults, dispatchGetAlgorithms } = this.props;
+    const {
+      dispatchGetResults,
+      dispatchGetAlgorithms,
+      dispatchGetPipelines,
+    } = this.props;
     dispatchGetResults();
     dispatchGetAlgorithms();
+    dispatchGetPipelines();
+  }
+
+  componentDidUpdate({ results: prevResults }) {
+    const { results } = this.props;
+    if (!prevResults.equals(results)) {
```

Comment:
I don't think that `groupBy` clarifies the problem since the results do not always have the `pipelineExecutionId` property. Therefore, grouping by `pipelineExecutionId` will lead to stock all the simple results (from an execution of an algorithm) into a single array,  then we will need to spread this array into multiple single arrays to still have the same output. I don't think that it simplifies the problem since we still need to know when the spreading into single arrays has to happen.